test/conformance/testing/include/uur/unsupported.h

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2022-2023 Intel Corporation
+// Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+// See LICENSE.TXT
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef UR_CONFORMANCE_INCLUDE_UNSUPPORTED_H_INCLUDED
+#define UR_CONFORMANCE_INCLUDE_UNSUPPORTED_H_INCLUDED
+
+#include "uur/utils.h"
+#include <string>
+#include <vector>
+
+namespace uur {
+struct DeviceMatcher {
+    DeviceMatcher(ur_platform_backend_t backend,
+                  std::vector<std::string> deviceNames)
+        : backend(backend), deviceNames(deviceNames) {}
+
+    ur_platform_backend_t backend;
+    std::vector<std::string> deviceNames;
+};
+
+struct OpenCL : DeviceMatcher {
+    OpenCL(std::initializer_list<std::string> il)
+        : DeviceMatcher(UR_PLATFORM_BACKEND_OPENCL, {il.begin(), il.end()}) {}
+};
+
+struct LevelZero : DeviceMatcher {
```

Comment:
for test skipping purposes we will probably need to distinguish between the l0 and l0 v2 adapters

Reply:
Do we have a way to query this in the code? Might need to resort to having the build system tell us?

Reply:
I don't think v2 reports anything different identification wise, @pbalcer?